gm4_better_crossbows/data/better_crossbows/functions/shoot/salmon.mcfunction

```diff
@@ -0,0 +1,13 @@
+# @s = player using the crossbow
+
+#summon salmon at player's location
+summon minecraft:salmon ~ ~1.2 ~ {FromBucket:1,Tags:["gm4_cb_proj"],ActiveEffects:[{Id:10,Amplifier:10,Duration:200,ShowParticles:0b}]}
+
+#replace bucket of salmon in player's offhand with water bucket
+execute unless entity @s[gamemode=creative] run replaceitem entity @s[nbt={Inventory:[{Slot:-106b,id:"minecraft:salmon_bucket"}]}] weapon.offhand minecraft:water_bucket

```

Comment:
```suggestion
execute unless entity @s[gamemode=creative] run replaceitem entity @s weapon.offhand minecraft:water_bucket
```
This nbt check isn't required as the item was already checked in `check_projectile.mcfunction`